modules/tinymce/src/core/main/ts/api/dom/SelectorChanged.ts

```diff
@@ -29,32 +29,37 @@ export default (dom: DOMUtils, editor: Editor) => {
   let selectorChangedData: Record<string, SelectorChangedCallback[]>;
   let currentSelectors: Record<string, SelectorChangedCallback[]>;
 
+  const matches = (selector: string, nodes: Node[]): boolean =>
+    Arr.exists(nodes, (node) => dom.is(node, selector));
+
+  const getParents = (elem: Element): Element[] =>
+    dom.getParents(elem, null, dom.getRoot());
+
   return {
     selectorChangedWithUnbind: (selector: string, callback: SelectorChangedCallback): { unbind: () => void } => {
       if (!selectorChangedData) {
         selectorChangedData = {};
         currentSelectors = {};
 
         editor.on('NodeChange', (e) => {
-          const node = e.element, parents = dom.getParents(node, null, dom.getRoot()), matchedSelectors = {};
+          const node = e.element;
+          const parents = getParents(node);
+          const matchedSelectors = {};
 
           // Check for new matching selectors
           Tools.each(selectorChangedData, (callbacks, selector) => {
-            Tools.each(parents, (node) => {
-              if (dom.is(node, selector)) {
-                if (!currentSelectors[selector]) {
-                  // Execute callbacks
-                  Tools.each(callbacks, (callback) => {
-                    callback(true, { node, selector, parents });
-                  });
-
-                  currentSelectors[selector] = callbacks;
-                }
-
-                matchedSelectors[selector] = callbacks;
-                return false;
+            if (matches(selector, parents)) {
+              if (!currentSelectors[selector]) {
+                // Execute callbacks
+                Arr.each(callbacks, (callback) => {
+                  callback(true, { node, selector, parents });
```

Comment:
In the old code, it looks like `node` passed into the callback could be a parent but in the new code, it will always be the node from the event. Just wanted to check my understanding is correct?

Reply:
You're correct, it shouldn't be doing that... I'll get that fixed.

Reply:
Fixed... this is also why shadowed variables are bad :cry: